lib/dal/src/attribute/value/dependent_value_graph.rs

```diff
@@ -249,12 +229,86 @@ impl DependentValueGraph {
+            // DependentValuesUpdate job may have been removed from the snapshot between when the
+            // DVU job was created and when we're processing things now. This could happen if there
+            // are other modifications to the snapshot before the DVU job starts executing, as the
+            // job always operates on the current state of the change set's snapshot, not the state
+            // at the time the job was created.
+            if workspace_snapshot
+                .try_get_node_index_by_id(initial_id)
+                .await?
+                .is_none()
+            {
+                debug!(%initial_id, "missing node, skipping it in DependentValueGraph");
+                continue;
+            }
+
+            let node_weight = workspace_snapshot.get_node_weight_by_id(initial_id).await?;
+
+            match node_weight {
+                NodeWeight::AttributeValue(_) => {
+                    // We need to pre-process the initial values here to add any object
+                    // child values. We do this before the work loop instead of during so
+                    // that we only add the child values of the values we have been
+                    // explicitly asked to calculate the graph for, not any objects
+                    // encountered while walking the prototype chain
+                    let value: AttributeValueId = initial_id.into();
+                    let child_values =
+                        AttributeValue::all_object_children_to_leaves(ctx, value).await?;
+                    values.push(value);
+                    values.extend(child_values);
+                }
+                NodeWeight::Content(content_node_weight) => {
+                    match content_node_weight.content_address().into() {
+                        // If we are processing a secret, we don't want to add the secret itself.
+                        // We only want to add the direct dependent attribute values. We also need
+                        // to mark these values as needing to be processed because DVU will skip
+                        // processing the first set of independent values. In most cases, we want
+                        // that to happen. However, since the first set of independent values all
+                        // have a secret as their parent and not an attribute value, they need to be
+                        // processed in DVU.
+                        ContentAddressDiscriminants::Secret => {
+                            let direct_dependents =
+                                Secret::direct_dependent_attribute_values(ctx, initial_id.into())
+                                    .await
+                                    .map_err(Box::new)?;
+                            self.values_that_need_to_execute_from_prototype_function
+                                .extend(direct_dependents.clone());
+                            values.extend(direct_dependents);
+                        }
```

Comment:
This is where we determine "what attribute values directly dependent on this secret"